src/main/java/org/mamute/controllers/AnswerController.java

```diff
@@ -89,12 +87,18 @@ public void newAnswer(@Load Question question, MarkedText description, boolean w
         	answers.save(answer);
         	ReputationEvent reputationEvent = new ReputationEvent(EventType.CREATED_ANSWER, question, current);
         	reputationEvents.save(reputationEvent);
-    		if (isUserWithKarma) current.increaseKarma(reputationEvent.getKarmaReward());
-        	result.redirectTo(QuestionController.class).showQuestion(question, question.getSluggedTitle());
+
+    		if (isUserWithKarma) {
+				current.increaseKarma(reputationEvent.getKarmaReward());
+			}
+
+			result.redirectTo(QuestionController.class).showQuestion(question, question.getSluggedTitle());
 			notificationManager.sendEmailsAndInactivate(new EmailAction(answer, question));
 			if (watching) {
 				watchers.add(question, new Watcher(current));
 			}
+			List<Attachment> attachmentsLoaded = attachments.load(attachmentsIds);
+			answer.add(attachmentsLoaded);
```

Comment:
since this line modifies the answer, I would prefer to leave it close to `new Answer(...)`